src/secret-storage.ts

```diff
@@ -200,7 +202,9 @@ export interface SecretStorageCallbacks {
     ) => Promise<[string, Uint8Array] | null>;
 }
 
-interface SecretInfo {
+export type SecretInfoKey = Assignable<AccountDataEvents, SecretInfo>;
```

Comment:
Wouldn't it be easier/clearer/make more sense generally to define `SecretInfoKey` explicitly, and derive `AccountData` from that, rather than the other way around?

Reply:
That wouldn't be extensible by consumers of this library, only interfaces are extensible via declaration merging.